src/pages/tasks/TaskAssigneeSelectorModal.tsx

```diff
@@ -1,57 +1,48 @@
 /* eslint-disable es/no-optional-chaining */
+import type {RouteProp} from '@react-navigation/native';
 import {useRoute} from '@react-navigation/native';
-import lodashGet from 'lodash/get';
-import lodashPick from 'lodash/pick';
-import PropTypes from 'prop-types';
 import React, {useCallback, useMemo, useState} from 'react';
 import {View} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import {useBetas, usePersonalDetails, useSession} from '@components/OnyxProvider';
 import ScreenWrapper from '@components/ScreenWrapper';
 import SelectionList from '@components/SelectionList';
+import type {RadioItem, User} from '@components/SelectionList/types';
+import withCurrentUserPersonalDetails from '@components/withCurrentUserPersonalDetails';
+import type {WithCurrentUserPersonalDetailsProps} from '@components/withCurrentUserPersonalDetails';
 import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
 import useDebouncedState from '@hooks/useDebouncedState';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
-import compose from '@libs/compose';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as ReportUtils from '@libs/ReportUtils';
-import reportPropTypes from '@pages/reportPropTypes';
-import * as Task from '@userActions/Task';
+import type {TaskDetailsNavigatorParamList} from '@navigation/types';
+import * as TaskActions from '@userActions/Task';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {Report, Task} from '@src/types/onyx';
 
-const propTypes = {
+type TaskAssigneeSelectorModalOnyxProps = {
     /** All reports shared with the user */
-    reports: PropTypes.objectOf(reportPropTypes),
+    reports: OnyxCollection<Report>;
 
     /** Grab the Share destination of the Task */
-    task: PropTypes.shape({
-        /** Share destination of the Task */
-        shareDestination: PropTypes.string,
-
-        /** The task report if it's currently being edited */
-        report: reportPropTypes,
-    }),
-
-    /** The policy of root parent report */
-    rootParentReportPolicy: PropTypes.shape({
-        /** The role of current user */
-        role: PropTypes.string,
-    }),
+    task: OnyxEntry<Task>;
 };
 
-const defaultProps = {
-    reports: {},
-    task: {},
-    rootParentReportPolicy: {},
+type UseOptionsProps = {
```

Comment:
nab: but shouldn't we name it as just `UseOptions` ? because `Props` suffix should be reserved only for components